src/arithmetic.jl

```diff
@@ -1,90 +1,78 @@
 # Add
-Base.:+(sv1::SecureVector, sv2::SecureVector) = add(sv1, sv2)
-Base.:+(sv::SecureVector, pv::PlainVector) = add(sv, pv)
-Base.:+(pv::PlainVector, sv::SecureVector) = add(sv, pv)
-Base.:+(sv::SecureVector, scalar::Real) = add(sv, scalar)
-Base.:+(scalar::Real, sv::SecureVector) = add(sv, scalar)
+Base.:+(sa1::SecureArray, sa2::SecureArray) = add(sa1, sa2)
```

Comment:
I think you probably want to be more restrictive here. Adding two arbitrary `SecureArray`s likely does not make sense unless they are both of matching configuration.

I know that we didn't check e.g. that the size of vectors or matrices fit before (length or capacity matching). However, at least we could be sure that nobody was able to try to add a matrix to a vector. Thus maybe we should at least enforce this here (since we can do this in type domain), e.g., with something like
```suggestion
Base.:+(sa1::SecureArray{B, N}, sa2::SecureArray{B, N}) where {B, N} = add(sa1, sa2)
```
This at least ensures that `B` and `N` need to be identical. This of course applies to all other functions in this file as well.